timeseries/src/autogluon/timeseries/models/ensemble/greedy_ensemble.py

```diff
@@ -1,226 +1,107 @@
-
-    def _calculate_regret(self, y_true, y_pred_proba, metric, dummy_pred, sample_weight=None):  # noqa
-        dummy_pred = copy.deepcopy(dummy_pred)
+    def _calculate_regret(self, y_true, y_pred_proba, metric, dummy_pred=None, sample_weight=None):  # noqa
+        dummy_pred = copy.deepcopy(self.dummy_pred if dummy_pred is None else dummy_pred)
         dummy_pred[list(dummy_pred.columns)] = y_pred_proba
-        score = metric(y_true, dummy_pred)
-        if self.higher_is_better:
-            score = -score
-        return score
+        return metric(y_true, dummy_pred) * metric.coefficient
 
 
 class SimpleTimeSeriesWeightedEnsemble(AbstractWeightedEnsemble):
     """Predefined user-weights ensemble"""
 
-    def __init__(self, weights, **kwargs):
+    def __init__(self, weights: List[float], **kwargs):
         self.weights_ = weights
+        self.problem_type = ag.constants.QUANTILE
 
     @property
     def ensemble_size(self):
         return len(self.weights_)
 
-    def predict(self, X):
-        return self.predict_proba(X)
-
-    def predict_proba(self, X):
-        return self.weight_pred_probas(X, weights=self.weights_)
-
     def weight_pred_probas(
         self,
         preds: List[TimeSeriesDataFrame],
         weights: List[float],
     ) -> TimeSeriesDataFrame:
-        # TODO: this is a hack. indices may not match, which should be checked or better,
-        # TODO: weighted accordingly
         assert len(set(v.shape for v in preds)) == 1
 
         # TODO: handle NaNs
         return sum(p * w for p, w in zip(preds, weights))
 
 
 class TimeSeriesEnsembleWrapper(AbstractTimeSeriesModel):
-    """Predefined user-weights ensemble"""
-
-    def __init__(self, weights, name, **kwargs):
+    """AbstractTimeSeriesModel wrapper for simple weighted ensemble selection
+    models.
+
+    Parameters
+    ----------
+    weights : Dict[str, float]
+        Dictionary mapping model names to weights in the ensemble
+    name : str
+        Name of the model. Usually like ``"WeightedEnsemble"``.
+    """
+
+    def __init__(self, weights: Dict[str, float], name: str, **kwargs):
         super().__init__(name=name, **kwargs)
-        self.model = SimpleTimeSeriesWeightedEnsemble(weights=weights)
-        self.name = name
-        self.fit_time = None
-        self.val_score = None
+        self.model_names, model_weights = zip(*weights.items())
+        self.weighted_ensemble = SimpleTimeSeriesWeightedEnsemble(weights=model_weights)
 
-    def _fit(
-        self,
-        train_data,
-        val_data=None,
-        time_limit=None,
-        num_cpus=None,
-        num_gpus=None,
-        verbosity=2,
-        **kwargs,
-    ) -> None:
+    def _fit(self, *args, **kwargs) -> None:
         raise NotImplementedError
 
     @property
     def weights(self):
-        return self.model.weights_
+        return self.weighted_ensemble.weights_
+
+    def predict(self, data: Dict[str, TimeSeriesDataFrame], **kwargs):
```

Comment:
The signature of `predict` is different from the signature of `AbstractTimeSeriesModel`. Should we call this function differently (e.g., `predict_ensemble` or `combine_predictions`)?

Reply:
Also, could we add a pointer here (or somewhere else) that explains the format of the predictions made by each individual model. As far as I understand, this `TimeSeriesDataFrame` is quite different from the usual dataset format.

Reply:
for the first issue, I made the abstract model method signature more generic for now. I don't think it's a perfect solution but it's also not part of the public API.